iocore/net/SSLConfig.cc

```diff
@@ -456,85 +455,38 @@ SSLConfigParams::initialize()
   client_ctx = SSLInitClientContext(this);
   if (!client_ctx) {
     SSLError("Can't initialize the SSL client, HTTPS in remap rules will not function");
-  } else {
-    InsertCTX(this->clientCertPath, this->client_ctx);
   }
 }
 
-// getCTX: returns the context attached to the given certificate
-SSL_CTX *
-SSLConfigParams::getCTX(cchar *client_cert) const
-{
-  ink_mutex_acquire(&ctxMapLock);
-  auto client_ctx = ctx_map.get(client_cert);
-  ink_mutex_release(&ctxMapLock);
-  return client_ctx;
-}
-
-// InsertCTX hashes on the absolute path to the client certificate file and stores in the map
-bool
-SSLConfigParams::InsertCTX(cchar *client_cert, SSL_CTX *cctx) const
-{
-  ink_mutex_acquire(&ctxMapLock);
-  // dup is required here to avoid the nullifying of the keys stored in the map.
-  // client_cert is coming from the overridable clientcert config retrieved by the remap plugin.
-  cchar *cert = ats_strdup(client_cert);
-  // Hashmap has no delete functionality :(
-  ctx_map.put(cert, cctx);
-  ink_mutex_release(&ctxMapLock);
-  return true;
-}
-
-void
-SSLConfigParams::printCTXmap() const
-{
-  Vec<cchar *> keys;
-  ctx_map.get_keys(keys);
-  for (size_t i = 0; i < keys.length(); i++) {
-    Debug("ssl", "Client certificates in the map %s: %p", keys.get(i), ctx_map.get(keys.get(i)));
-  }
-}
-void
-SSLConfigParams::freeCTXmap() const
-{
-  ink_mutex_acquire(&ctxMapLock);
-  Vec<cchar *> keys;
-  ctx_map.get_keys(keys);
-  size_t n = keys.length();
-  Debug("ssl", "freeing CTX Map");
-  for (size_t i = 0; i < n; i++) {
-    deleteKey(keys.get(i));
-    ats_free((char *)keys.get(i));
-  }
-  ctx_map.clear();
-  ink_mutex_release(&ctxMapLock);
-}
 // creates a new context attaching the provided certificate
 SSL_CTX *
-SSLConfigParams::getNewCTX(cchar *client_cert) const
+SSLConfigParams::getNewCTX(cchar *client_cert, cchar *client_key) const
 {
   SSL_CTX *nclient_ctx = nullptr;
   nclient_ctx          = SSLInitClientContext(this);
   if (!nclient_ctx) {
     SSLError("Can't initialize the SSL client, HTTPS in remap rules will not function");
     return nullptr;
   }
-  if (nclient_ctx && client_cert != nullptr && client_cert[0] != '\0') {
-    if (!SSL_CTX_use_certificate_chain_file(nclient_ctx, (const char *)client_cert)) {
-      SSLError("failed to load client certificate from %s", this->clientCertPath);
-      SSLReleaseContext(nclient_ctx);
-      return nullptr;
+  if (nclient_ctx) {
+    if (client_cert != nullptr && client_cert[0] != '\0') {
+      if (!SSL_CTX_use_certificate_chain_file(nclient_ctx, (const char *)client_cert)) {
+        SSLError("failed to load client certificate from %s", this->clientCertPath);
+        SSLReleaseContext(nclient_ctx);
+        return nullptr;
+      }
+    }
```

Comment:
Shouldn't this be an `else`?

Reply:
No,  the previous clause is checking client_cert.  This clause is checking client_key.

But the outer check on nclient_ctx is unnecessary since we already returned in the case of that being null.